spec/ics-027-interchain-account/README.md

```diff
@@ -0,0 +1,285 @@
+---
+ics: 27  
+title: Interchain Account  
+stage: Draft  
+category: IBC/TAO  
+author: Tony Yun <yunjh1994@everett.zone>, Dogemos <josh@tendermint.com>  
+created: 2019-08-01  
+modified: 2019-11-08  
+---  
+
+## **Synopsis**  
+
+This standard document specifies packet data structure, state machine handling logic, and encoding details for the account management system over an IBC channel between separate chains.  
+
+### **Motivation**  
+
+On Ethereum, there are two types of accounts: externally owned accounts, controlled by private keys, and contract accounts, controlled by their contract code [[ref](https://github.com/ethereum/wiki/wiki/White-Paper)]. Similar to Ethereum's CA(contract accounts), Interchain accounts are managed by another chain while retaining all the capabilities of a normal account (i.e. stake, send, vote, etc). While an Ethereum CA's contract logic is performed within Ethereum's EVM, Interchain accounts are managed by another chain via IBC in a trustless way.  
+
+### **Definitions**  
+
+The IBC handler interface & IBC relayer module interface are as defined in [ICS 25](https://github.com/cosmos/ics/blob/bez/15-ics-cosmos-signed-messages/spec/ics-025-handler-interface) and [ICS 26](https://github.com/cosmos/ics/blob/bez/15-ics-cosmos-signed-messages/spec/ics-026-relayer-module), respectively.  
+
+### **Desired Properties**  
+
+- Permissionless  
+- Fault containment: Interchain account must follow rules of its host chain, even in times of byzantine behavior by the counterparty chain (the chain that manages the account)  
+- The chain that controls the account must process the results asynchronously and according to the chain's logic.  
+- Sending and receiving transactions will be processed in an ordered manner.  
+
+## **Technical Specification**  
+
+The implementation of interchain account is non symmetric. This means that each chain must define the way to create a interchain account for counterparty chain deterministically.
```

Comment:
Also I'm a little confused. Does different logic have to be defined for each counterparty chain? Do you mean that the transaction bytes will be different?

Reply:
It means that each chain can have a different way to deserialize transaction bytes and a different set of transactions that they can execute. For example, chains that use cosmos-sdk will deserialize tx bytes by amino, but if counterparty chain is a smart contract on Ethereum, it may deserialize tx bytes by ABI that is minimal serialize algorithm for smart contract. And a smart contract on Ethereum can't execute the staking, governance transaction if it doesn't support such feature. So, I think interchain account defines the general way to register interchain account and transfer tx bytes, and counterparty chain is responsible to deserialize and execute the tx bytes, and sending chain should know how counterparty chain handle tx bytes in advance.

Reply:
Makes sense, just include the above paragraph in the spec so that it's clear.